src/components/Articles.jsx

```diff
@@ -1,23 +1,29 @@
 import { useEffect, useState } from "react";
-import { Link } from "react-router-dom";
+import { Link, useParams } from "react-router-dom";
 import { getReq } from "../utils/api";
 
 const Articles = () => {
+  const { topic_name } = useParams();
   const [articleList, setArticleList] = useState([]);
   const [err, setErr] = useState(null);
 
   useEffect(() => {
     const asyncEffect = async () => {
       try {
         const result = await getReq("/articles");
-        setArticleList(result);
-        console.log(result);
+        if (topic_name) {
+          setArticleList(() => {
+            return result.filter((article) => article.topic === topic_name);
+          });
+        } else {
+          setArticleList(result);
+        }
```

Comment:
You can use an API call here to grab only the topics you are looking for (e.g if (topic_name){  path += `/?topic=${topic_name}`} etc) rather than a filter